src/DocBlock/TypeExpression.php

```diff
@@ -141,6 +141,11 @@ public function __construct(string $value, ?NamespaceAnalysis $namespace, array
         $this->parse();
     }
 
+    public function isUnionType(): bool
+    {
+        return $this->isUnionType && '|' === $this->getTypesGlue();
```

Comment:
does this mean that `isUnionType property` and `isUnionType getter method` are representing 2 different concepts?

Reply:
Well.. it is some time since I made this PR, so I might be mistaken, but as far as I remember - yes, they are a "different" thing

`isUnionType` property is `true` for both `union` and `intersection` types but the method `isUnionType` is really only for a `union` type

Reply:
i think we should clean it up then. having same name for 2 different meanings is confusing. I guess original property was created before intersection was a thing in PHP

Reply:
Yes I agree. 
But I'm not sure about a correct name
- `isUnionOrIntersectionType`
- `hasMultipleTypes`
- different?

Reply:
FYI: `isUnionType()` was added in the meantime by @mvorisek during improvements for phpDoc parsing, so I removed implementation from current PR and moved `'|' === $this->getTypesGlue()` part to place where `isUnionType()` is invoked. Behaviour is the same, just there's no inconsistency between the method and property. Naming should be improved though (because "union" in this context means "complex" - both union or intersection), but not in this PR.